File: mobile/android/FUN/app/src/main/java/com/fun/app/data/models/Unlock.kt
/**
 * FUN App - Unlock Models
 */

package com.fun.app.data.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Unlock(
    @Json(name = "_id") val id: String,
    val seriesId: String,
    val episodeNum: Int,
    val method: String,  // "ad", "credits", "iap", "premium"
    val creditsSpent: Int?,
    val unlockedAt: String
)

@JsonClass(generateAdapter = true)
data class UnlockRequest(
    val seriesId: String,
    val episodeNum: Int,
    val method: String,
    val adProof: String? = null,
    val receipt: String? = null
)

@JsonClass(generateAdapter = true)
data class UnlockResponse(
    val unlock: Unlock,
    val message: String
)

@JsonClass(generateAdapter = true)
data class UnlocksResponse(
    val unlocks: List<Unlock>
)

File: mobile/ios/FUN/FUN/Models/Unlock.swift
/**
 * FUN App - Unlock Model
 */

import Foundation

struct Unlock: Codable, Identifiable {
    let id: String
    let seriesId: String
    let episodeNum: Int
    let method: String  // "ad", "credits", "iap", "premium"
    let creditsSpent: Int?
    let unlockedAt: String
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case seriesId, episodeNum, method, creditsSpent, unlockedAt
    }
}

struct UnlockRequest: Codable {
    let seriesId: String
    let episodeNum: Int
    let method: String
    let adProof: String?
    let receipt: String?
}

struct UnlockResponse: Codable {
    let unlock: Unlock
    let message: String
}

struct UnlocksResponse: Codable {
    let unlocks: [Unlock]
}
